addons/io_scene_gltf2/__init__.py

```diff
@@ -429,6 +441,9 @@ def execute(self, context):
         export_settings['gltf_binaryfilename'] = os.path.splitext(os.path.basename(
             bpy.path.ensure_ext(self.filepath,self.filename_ext)))[0] + '.bin'
 
+        user_extensions = getattr(self, 'user_extensions', [])
```

Comment:
Okay, okay! I figured out why you're doing it this way :) `invoke` only gets called when a human uses the operator; if it's done from a script, like the testing script for the CI, only `execute` gets called, right?

Obviously we still want to use user-extensions if the exporter is called from a script though, so I think this should be like this...

In invoke, do the panel stuff:

````py
# Register extension panels
import sys
for addon_name in bpy.context.preferences.addons.keys():
    try:
        if not hasattr(sys.modules[addon_name], 'glTF2UserExtension'):
            continue
        sys.modules[addon_name].register_panel()
    except Exception:
        pass
````

Here in execute, collect the extension objects:

````py
import sys
user_extensions = []
for addon_name in bpy.context.preferences.addons.keys():
    if not hasattr(sys.modules[addon_name], 'glTF2UserExtension'):
        continue
    user_extensions.append(sys.modules[addon_name].glTF2UserExtension())
export_settings['gltf_user_extensions'] = user_extensions
````

Reply:
Done

Reply:
It looks like `invoke` didn't change though.